### ISSUE TITLE
Random rendered features have mismatched properties

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-gl-native#5182]
Random features returned by `mbgl::Map::queryRenderedFeatures()` have mismatched or missing properties.

When the current style is Mapbox Streets (which uses Mapbox Streets source), calling `mbgl::Map::queryRenderedFeatures()` with the location of [this OSM node](http://www.openstreetmap.org/node/154075308) at z14 returns a point feature in which the `scalerank` and `type` are conflated. Here’s the output from `MGLPointFeature.attributes`:

```plist
{
    ldir = E;
    localrank = 1;
    name = Loveland;
    "name_de" = Loveland;
    "name_en" = Loveland;
    "name_es" = Loveland;
    "name_fr" = Loveland;
    "name_ru" = "\U041b\U0430\U0432\U043b\U0435\U043d\U0434";
    "name_zh" = "\U6d1b\U5f17\U5170\U5fb7";
    scalerank = town;
}
```

In GL JS v0.18.0, `queryRenderedFeatures()` doesn’t mismatch any tags, but `scalerank` is nowhere to be seen:

```js
{
  ldir: "E",
  localrank: 1,
  name: "Loveland",
  name_de: "Loveland",
  name_en: "Loveland",
  name_es: "Loveland",
  name_fr: "Loveland",
  name_ru: "Лавленд",
  name_zh: "洛弗兰德",
  type: "town"
}
```

Querying rendered features at 39°5′43″ north, 84°31′20″ west (in the 13/2172/3128 tile) yields several features in which the `layer`, `oneway`, `structure`, and `type` properties are all off by one:

```plist
{
    class = motorway;
    layer = true;
    oneway = bridge;
    structure = motorway;
    type = 2;
}
```

Compare with GL JS:

```js
{
  class: "motorway",
  layer: 2,
  oneway: "true",
  structure: "bridge",
  type: "motorway"
}
```

/cc @jfirebaugh

**Comment by 1ec5** [comment]
This issue affects all the features in an affected tile, except for rendered symbol features, which appear to be stored in a separate tree.

Here’s `VectorTileLayer::keysMap` for the `place_label` tile layer in the first example above:

```
keysMap = size=11 {
  [0] = (first = "ldir", second = 0)
  [1] = (first = "localrank", second = 1)
  [2] = (first = "name", second = 2)
  [3] = (first = "name_de", second = 3)
  [4] = (first = "name_en", second = 4)
  [5] = (first = "name_es", second = 5)
  [6] = (first = "name_fr", second = 6)
  [7] = (first = "name_ru", second = 7)
  [8] = (first = "name_zh", second = 8)
  [9] = (first = "scalerank", second = 10)
  [10] = (first = "type", second = 9)
}
```

**Comment by 1ec5** [comment]
I’m still not sure what’s going on w/ the `scalerank` key. I’d expect it to be present in the tiles, and this bug uncovered its existence, but the value is missing from both GL JS’s parsed tags and mbgl’s parsed tags with the fix in #5183.
